spec/queries/product_scope_query_spec.rb

```diff
@@ -101,11 +101,24 @@
 
     it "finds products by producer" do
       # Add variants so we can check if we are not returning duplicate products
-      create(:variant, product: product3, supplier: producer)
-      create(:variant, product: product3, supplier: producer)
+      create(:variant, product: product3, supplier: producer_z)
+      create(:variant, product: product3, supplier: producer_z)
 
       expect(query.products_for_producers).to eq([product3, product4])
     end
+
+    it "order products by producer alphabetically and product alphabetically" do
+      producer_b = create(:supplier_enterprise, name: "b_name")
+      product_b1 = create(:product, name: "g_name", supplier_id: producer_b.id)
+      product_b2 = create(:product, name: "z_name", supplier_id: producer_b.id)
+      product_b3 = create(:product, name: "a_name", supplier_id: producer_b.id)
+
+      create(:enterprise_relationship, parent: producer_b, child: hub,
+                                       permissions_list: [:create_variant_overrides])
+
+      result = query.products_for_producers
+      expect(result).to eq([product_b3, product_b1, product_b2, product3, product4])
```

Comment:
At first glance, it's not easy to see if this is correct or not. We could name the variables the same as the record names to make it clearer.

Not that I'm complaining, at least we have a spec now!!